src/targets/gpu/jit/pointwise.cpp

```diff
@@ -131,12 +143,20 @@ struct pointwise_compiler : compiler<pointwise_compiler>
         auto preloads          = preload(axis, options.virtual_inputs);
         auto is_preloading =
             std::accumulate(preloads.begin(), preloads.end(), false, std::logical_or<>{});
+        auto op_names              = v.at("op_names").to_vector<std::string>();
+        std::string op_name_string = join_strings(op_names, "_");
+        if(op_name_string.empty())
+            op_name_string = "kernel";
+        else
+            op_name_string += "_kernel";
+        options.kernel_name = op_name_string;
         options.set_launch_params(v,
                                   compute_global_for(ctx,
                                                      options.output.elements() / vec_size,
                                                      oversubscribe_if(not is_preloading)));
         auto src = interpolate_string(pointwise_kernel,
-                                      {{"params", enum_params(inputs.size(), "void * private_p")},
+                                      {{"op_names", op_name_string},
```

Comment:
Just name the key `kernel` such as `{"kernel", v.get("kernel", "kernel")}`.

Reply:
And set the kernel name with `options.kernel_name = v.get("kernel", "kernel")` with the first `"kernel"` being the key and the second is the fallback value if its missing.

The reason for the fallback is that we may not pass this in when running from the gpu-driver.